Raster styler2

jorgenevil opened the pull request:
Begynnelsen på raster styler. Litt buggy, men oppdaterer cube rastere _hvis de ikke har vært laget før_ ... dvs, hvis det allerede finnes et pregenerert raster, vil den bruke dette snarere enn å lage nye filer. (Se bilde nedenfor, hvor flere stylinger henger igjen)

![screen shot 2016-04-04 at 2 22 02 pm](https://cloud.githubusercontent.com/assets/8961926/14247876/8b2c86f0-fa71-11e5-9596-b6fd5a789a4c.png)

Commit:
beginning of raster styler

Commit:
merge fix

Commit:
cleanup

Commit:
Merge branch 'master' into rasterStyler

Commit:
raster styler continued

knutole commented:
@jorgenevil Har du en PR for `wu` også? (app.ejs og Gruntfile?)

knutole committed:
Add cache value to url for hard-refresh of layer after styling

knutole commented:
Cached-tiles issue fixed in 489b8b9.

knutole committed:
Open BigSlider when selecting Cube layer

knutole committed:
Fixed 'Failed to parse SourceMap' issue with dc.js

Commit:
merge fix

Commit:
fixes 192

knutole committed:
Fixed 'Merged issue/192

knutole commented:
@jorgenevil har merga https://github.com/systemapic/systemapic.js/pull/195 inn i denne PR'en. så har vi alt på ett sted ift slider/styler. Kan bare `git checkout rasterStyler2 && git pull` og fortsette der (så får du med commitsa mine også.)

knutole commented:
@jorgenevil Ble veldig nice med slider. og dragging :+1:

knutole committed:
Added styleJSON saving for Wu.Cubelayer

Commit:
working on styler

Commit:
merge fix

Commit:
save raster styling

Commit:
added opacity support for styling

knutole committed:
Add list of datasets contained in cube, with edit options.
- todo: sortable, update, create-cube-button, etc.
- todo: Create better metadata for datasets in cube

knutole committed:
Add sortable dataset-list for Cubelayer.
- todo: update cube on changes, add/remove datasets, more options

knutole committed:
Add/remove datasets from Cube in layer options

knutole committed:
Create cube layer from Upload-button options.
- bugs: can't add to layermeny when empty

knutole committed:
Moved chart/slider to bottom-left corner

knutole committed:
UNDO: oved chart/slider to bottom-left corner

knutole committed:
animationSlider event

knutole committed:
Remove blue line on slider

knutole committed:
Indentation: 4/spaces

knutole committed:
Indentation: 4/spaces

knutole committed:
Set title of layer on bigSlider

knutole committed:
Update layer when add/removing datasets to cube

knutole committed:
Remove console log

knutole committed:
Open 'Add dataset'-dropdown by default

knutole committed:
Renamed Wu.PostGISLayer -> Wu.VectorLayer

knutole committed:
Comments, cleanup

knutole committed:
fixed text overflow on layer menu

knutole committed:
minor cleanup

Commit:
some stuff...

knutole committed:
Updated noUISlider with multiple-handles PR
- see https://github.com/leongersen/noUiSlider/pull/592

knutole commented:
@jorgenevil Har lagt til noUISlider versjonen som støtter mulitple handles, se commit 0e388df
 over. Mulig det er noen bugs, og har sett at blant annet [`limit` option](https://github.com/systemapic/systemapic.js/blob/rasterStyler2/js/src/panes/pane.bigslider.js#L247) kan ikke brukes. 

(Se her https://github.com/leongersen/noUiSlider/pull/592 for PR'en vi har merget.)

knutole committed:
Added noUISlider (need to rename)

knutole committed:
Cleanup of layer.isRaster(), .isVector() etc.

Commit:
separating slider and graph

Commit:
merge fix

knutole committed:
Cleanup of raster.styler.js (names, opacity=0 bugfix)

knutole committed:
model.layers.js merge fix

knutole committed:
Added styler for single-raster
- todo: save the style to pile/wu

knutole committed:
Renamed BigSlider -> Wu.Animator, BigSliderGraph -> Wu.Graph.Year

knutole committed:
Saving for single-raster. Bugfixed stops2cartocss (must add upper limit also)

knutole committed:
minor cleanup

Commit:
beginning to support multiple stops

knutole committed:
cleanup

knutole committed:
Added pikaday.js

knutole committed:
Added pikaday to cube time

knutole committed:
added pikaday.css

knutole committed:
Implemented manual date editing for Cube datasets

knutole committed:
Arrow down on layers (css)

knutole committed:
Minor css tweaks, comments

knutole committed:
Added Leaflet 1.0.0-beta2
- not plugged in, breaking changes

Commit:
multihandle slider

Commit:
merge fix

Commit:
minor fix

jorgenevil committed:
slider prider

knutole committed:
Upgrade Leaflet to 1.0.0-beta2, yohoo!
- lots of improvements: https://github.com/Leaflet/Leaflet/blob/master/CHANGELOG.md

jorgenevil committed:
raster slider with list

jorgenevil committed:
Merge branch 'rasterStyler2' of github.com:systemapic/systemapic.js into rasterStyler2

jorgenevil committed:
raster slider with list => done

jorgenevil committed:
tiny adjustments

knutole committed:
Remove all cube layers when removing layer

knutole committed:
Optimize javascript
- make sure `_refresh()` or `_refreshFiles()` is not called when not necessary
- optimized: add layer to layermeny, rename layer, onLayerEdited
- see https://github.com/systemapic/systemapic.js/issues/203

knutole committed:
Comments on rewriting chrome.data.js _refresh()
- see https://github.com/systemapic/systemapic.js/issues/203

knutole committed:
Added skim support for raster blur

knutole committed:
Separated files for layers, moved d3list to ext

knutole committed:
Fixed missing ; in blur css

knutole committed:
fixed typo

knutole committed:
Refactor of Cube layer animation begun
- cursor for current frame
- need to bring graph, slider, cube-layer together for animation

knutole committed:
Plugged date-event, connecting animator, graph and cube by today's date.
- todo: move cursor to today's date
- todo: organize datasets as frames
- todo: make cache work (incl. destroy)

knutole committed:
context bug

knutole committed:
Animation cache - work in progress

knutole committed:
Updated with new dataset scheme

knutole committed:
Layer updates on dragging animator

jorgenevil committed:
dropdown fixes + styler

jorgenevil committed:
Merge branch 'rasterStyler2' of github.com:systemapic/systemapic.js into rasterStyler2

jorgenevil committed:
fixes nicer drop down list -- issue #206

jorgenevil committed:
fixes placeholde/values in datalibrary -- issue #206

knutole committed:
Add timers for perf checks

knutole committed:
Cache queue implemented for cube tiles
- recycled Leaflet layers
- cacheSize and fps options
- todo: check timeline for memory leaks

knutole committed:
Removed FPS from cube layer
- only animator that controls fps
- cleanup

knutole committed:
JS Profile: optimized dataset search, 20% faster cache lookup

knutole committed:
JS Profile: optimized dataset index

knutole committed:
Improved flow of cache, animator, graph.
- todo: lots of improvements possible still, especially with Leaflet and sockets
- check: can graph drawing be improved more? #200
- timeline: need to check properly with a timeline

knutole committed:
Updated Leaflet.js to 1.0.0-rc1

knutole committed:
Added more-experiments.css

knutole committed:
Added layer load event; moved mixin listener to _initLayer() to avoid unnecessary listening on inactive layers

knutole committed:
Experimenting with map options; todo: fix bug on layer cache events

knutole committed:
Improved cache algorithm; cleaned up. Animator good enough for now.

knutole committed:
Animator erformance: 3fps

jorgenevil committed:
fixes #207

knutole committed:
Leaflet src change

knutole committed:
Leaflet line-endings (unix)

knutole commented on js/src/core/app.js:
```diff
@@ -410,18 +412,34 @@ Wu.App = Wu.Class.extend({
+		if(window.innerHeight > window.innerWidth) var portrait = true;
+
+		// Get width
+		var width = portrait ? app.isMobile.width : app.isMobile.height;
```
@jorgenevil `portrait` er `undefined` her hvis ikke portrait=true. Må definere `var portrait` utenfor if/else.

knutole committed:
Improvements on cache
- loading bug still active, but not important (fix with #210)
- remove stale model.layers.cube.js

jorgenevil committed:
fixes #208

jorgenevil committed:
Merge branch 'rasterStyler2' of github.com:systemapic/systemapic.js into rasterStyler2

knutole committed:
Added some Leaflet plugins (experiments)

jorgenevil committed:
div small stuff

jorgenevil committed:
fixing some graph bugs

jorgenevil committed:
idk

knutole committed:
Update leaflet.css; add map click event; cleanup; timings

knutole committed:
remove tourettes

knutole committed:
Remove some major rendering bottlenecks in DOM

jorgenevil committed:
raster styler fixes

jorgenevil committed:
Merge branch 'rasterStyler2' of github.com:systemapic/systemapic.js into rasterStyler2

knutole committed:
Undo adding leaflet.css 1.0.0-rc1

jorgenevil committed:
raster styler range support

knutole committed:
Add infinite.js to lib

knutole committed:
DEBUG: _.sample() data in dataset list & cube list
- also: beginning work on GUI for cube mask
- lots of small things

knutole committed:
Add TopoJSON as mask layer to cube; update Leaflet css til 1.0.0; add query route for geometry features

knutole committed:
css

knutole committed:
Query cube for SCF with GeoJSON mask
- todo: mark polygon for active query
- todo: when moving cursor after querying polygon, query for polygon only
- tood: lots of fixes, improvements

knutole committed:
Queries on polygons; graph css; optimizations
- pre total refactor of graph.year.js

knutole committed:
Refactored graph.annual.js; query either whole or parts of mask; lots n lots of other things

knutole committed:
cleanup
